### PR TITLE
First pass implementation of a re-index command

### PR DESCRIPTION
**Pull request by portante**
The command is intended to be run interactively on a pbench-server, not via a service.

We add a new cronjob, `pbench-re-index`, which only looks for re-indexing work under the new `TO-RE-INDEX` controller directory.  This will allow re-indexing work to happen independently of any new indexing work.

### CODE REVIEW

File: server/bin/pbench-reindex.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+"""Pbench Re-Index
+
+Re-index all data for a given date range, "YYYY-MM-DD:YYYY-MM-DD".
+
+The process of re-indexing first looks to see which INDEXED, TO-INDEX-TOOL,
+and WONT-INDEX* symlinks exist in the given date range, and moves them to
+TO-RE-INDEX.
+
+NOTE: this interface is intented to be used interactively, this is NOT a
```
Comment (ndokos) [inline server/bin/pbench-reindex.py:12]: 'intended'.

File: server/bin/pbench-reindex.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+        print(f"{act_set!r}")
+
+    print(f"Run-time: {start} {end} {end - start}")
+    return 0
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser(f"Usage: {_NAME_} [--config <path-to-config-file>]")
+    parser.add_argument("-C", "--config", dest="cfg_name", help="Specify config file")
+    parser.set_defaults(cfg_name=os.environ.get("_PBENCH_SERVER_CONFIG"))
```
Comment (ndokos) [inline server/bin/pbench-reindex.py:244]: Incorporate the default setting into the `.add_argument()` call?

File: server/bin/pbench-reindex.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+            f"'{exc}', expected time range values in the form YYYY-MM-DD",
+            file=sys.stderr,
+        )
+        return 7
+    else:
+        if newest_dt < oldest_dt:
+            # For convenience, swap oldest and newest dates that are reversed.
+            oldest_dt, newest_dt = newest_dt, oldest_dt
+
+    print(f"Re-indexing tar balls in the range {oldest_dt} to {newest_dt}")
```
Comment (ndokos) [inline server/bin/pbench-reindex.py:222]: Where does stdout end up?

File: server/bin/pbench-reindex.py [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+
+    actions = []
+    start = pbench._time()
+    for _val in gen_reindex_list(archive_p, oldest_dt, newest_dt):
+        controller_name, tb_name = _val
```
Comment (ndokos) [inline server/bin/pbench-reindex.py:227]: Can't the assignment be in the for loop, eliminating `_val`?

File: server/bin/pbench-reindex.py [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+                f"WARNING: {linkname_p.parent.name} link already exists for"
+                f" {controller_p / tb_name}"
+            )
+            # Indicate no action taken, and exit early.
+            return (controller_name, tb_name, "exists", msg)
+        elif linkname_p.parent.name in _linkdirs:
+            # One of the expected symlinks, add it for consideration below.
+            paths.append(linkname_p)
+        elif linkname_p.parent.name.startswith("WONT-INDEX"):
+            # One of the expected WONT-INDIX* symlinks, also added for
```
Comment (ndokos) [inline server/bin/pbench-reindex.py:75]: 'WONT-INDEX'

File: server/bin/pbench-index.py [hunk 6]
```diff
@@ -544,14 +550,22 @@ def main(options, name):
...
     )
+    parser.set_defaults(index_tool_data=False)
+    parser.add_argument(
+        "-R",
+        "--re-index",
+        action="store_true",
+        dest="re_index",
+        help="Perform re-indexing of previously indexed data",
+    )
+    parser.set_defaults(re_index=False)
```
[the remark this answers is not in the document]
Comment (portante) [reply server/bin/pbench-index.py:568]: Okay.

File: server/bin/pbench-reindex.py [hunk 7]
```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+"""Pbench Re-Index
+
+Re-index all data for a given date range, "YYYY-MM-DD:YYYY-MM-DD".
```
[the remark this answers is not in the document]
Comment (portante) [reply server/bin/pbench-reindex.py:6]: Not sure how this implies one argument, but I'll fix the comment to make it more "prose".

File: server/bin/pbench-reindex.py (hunk 1 above)
Comment (portante) [reply server/bin/pbench-reindex.py:12]: Fixed.

File: server/bin/pbench-reindex.py (hunk 2 above)
Comment (portante) [reply server/bin/pbench-reindex.py:244]: Fixed.

File: server/bin/pbench-reindex.py (hunk 3 above)
Comment (portante) [reply server/bin/pbench-reindex.py:222]: Console of where `pbench-reindex` is invoked.

File: server/bin/pbench-reindex.py (hunk 4 above)
Comment (portante) [reply server/bin/pbench-reindex.py:227]: Funny you would ask that.  Our beloved friend, `black`, handles the formatting of this construct so badly when I had that in the loop construct that it seemed based to format it this way.  An alternative would be to do:

```
_gen = gen_reindex_list(archive_p, oldest_dt, newest_dt)
for controller_name, tb_name in _gen:
   ...
```

File: server/bin/pbench-reindex.py (hunk 5 above)
Comment (portante) [reply server/bin/pbench-reindex.py:75]: Fixed.

File: server/bin/pbench-reindex.py (hunk 3 above)
Comment (ndokos) [reply server/bin/pbench-reindex.py:222]: Oh, right - I keep forgetting it's not a crontab script.

File: server/bin/pbench-reindex.py (hunk 4 above)
Comment (ndokos) [reply server/bin/pbench-reindex.py:227]: Fire black!

**Pull request merged**